core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/DistinctModelReducingUnionIteration.java

```diff
@@ -34,15 +34,15 @@
 public class DistinctModelReducingUnionIteration extends LookAheadIteration<Statement, SailException> {
 
 	DistinctModelReducingUnionIteration(CloseableIteration<? extends Statement, SailException> iterator, Model model,
-			Function<Model, Model> filterable) {
+			Function<Model, Iterator<Statement>> filterable) {
```

Comment:
Can we keep the old constructor signature as a (deprecated) overload? Just to keep things backward compatible.

Reply:
Actually never mind, I overlooked that this constructor is package-private anyway.